tests/device_test.sh

```diff
@@ -204,4 +204,102 @@ function test_get_desktop_environment()
+00:14.2 Signal processing controller: Intel Corporation Sunrise Point-LP Thermal subsystem (rev 21)
+00:15.0 Signal processing controller: Intel Corporation Sunrise Point-LP Serial IO I2C Controller #0 (rev 21)
+00:16.0 Communication controller: Intel Corporation Sunrise Point-LP CSME HECI #1 (rev 21)
+00:17.0 SATA controller: Intel Corporation Sunrise Point-LP SATA Controller [AHCI mode] (rev 21)
+00:1c.0 PCI bridge: Intel Corporation Sunrise Point-LP PCI Express Root Port #1 (rev f1)
+00:1c.4 PCI bridge: Intel Corporation Sunrise Point-LP PCI Express Root Port #5 (rev f1)
+00:1c.5 PCI bridge: Intel Corporation Sunrise Point-LP PCI Express Root Port #6 (rev f1)
+00:1f.0 ISA bridge: Intel Corporation Sunrise Point LPC Controller/eSPI Controller (rev 21)
+00:1f.2 Memory controller: Intel Corporation Sunrise Point-LP PMC (rev 21)
+00:1f.3 Audio device: Intel Corporation Sunrise Point-LP HD Audio (rev 21)
+00:1f.4 SMBus: Intel Corporation Sunrise Point-LP SMBus (rev 21)
+01:00.0 Display controller: Advanced Micro Devices, Inc. [AMD/ATI] Jet PRO [Radeon R5 M230 / R7 M260DX / Radeon 520 Mobile] (rev c3)
+02:00.0 Ethernet controller: Realtek Semiconductor Co., Ltd. RTL810xE PCI Express Fast Ethernet controller (rev 07)
+03:00.0 Network controller: Qualcomm Atheros QCA9565 / AR9565 Wireless Network Adapter (rev 01)
+EOF
+)
+
+mocked_lspci_vs1=$(
+  cat << EOF
+01:00.0 Display controller: Advanced Micro Devices, Inc. [AMD/ATI] Jet PRO [Radeon R5 M230 / R7 M260DX / Radeon 520 Mobile] (rev c3)
+  Subsystem: Dell Jet PRO [Radeon R5 M230 / R7 M260DX / Radeon 520 Mobile]
+  Flags: bus master, fast devsel, latency 0, IRQ 129
+  Memory at c0000000 (64-bit, prefetchable) [size=256M]
+  Memory at d0000000 (64-bit, non-prefetchable) [size=256K]
+  I/O ports at e000 [size=256]
+  Expansion ROM at d0040000 [disabled] [size=128K]
+  Capabilities: [48] Vendor Specific Information: Len=08 <?>
+  Capabilities: [50] Power Management version 3
+  Capabilities: [58] Express Legacy Endpoint, MSI 00
+  Capabilities: [a0] MSI: Enable+ Count=1/1 Maskable- 64bit+
+  Capabilities: [100] Vendor Specific Information: ID=0001 Rev=1 Len=010 <?>
+  Capabilities: [150] Advanced Error Reporting
+  Capabilities: [270] Secondary PCI Express
+  Kernel driver in use: radeon
+  Kernel modules: radeon, amdgpu
+EOF
+)
+
+mocked_lspci_vs2=$(
+  cat << EOF
+00:02.0 VGA compatible controller: Intel Corporation UHD Graphics 620 (rev 07) (prog-if 00 [VGA controller])
+  DeviceName:  Onboard IGD
+  Subsystem: Dell UHD Graphics 620
+  Flags: bus master, fast devsel, latency 0, IRQ 128
+  Memory at d1000000 (64-bit, non-prefetchable) [size=16M]
+  Memory at b0000000 (64-bit, prefetchable) [size=256M]
+  I/O ports at f000 [size=64]
+  Expansion ROM at 000c0000 [virtual] [disabled] [size=128K]
+  Capabilities: <access denied>
+  Kernel driver in use: i915
+  Kernel modules: i915
+EOF
+)
+
+function lspci_mock()
```

Comment:
After reviewing this commit, I'm under the impression that we only use the `else` part, and the `mocked_lspci_*` variables are not used. Did I miss something? If those variables are not used, please drop them.

Reply:
I might be wrong, but I believe that I'm testing whether the global variable `$gpus` was filled as expected, using the output from the two mocked `lspci -v -s` (`mocked_lspci_vs1` and `mocked_lspci_vs2`), i.e., I'm testing if the array `$gpus` has the values extracted from `"Subsystem: (.*)"` and "`controller: (.*)`".